soundcard/pulseaudio.py

```diff
@@ -369,7 +369,7 @@ def get_microphone(id, include_loopback=False, exclude_monitors=True):
         include_loopback = not exclude_monitors
 
     microphones = _pulse.source_list
-    return _Microphone(id=_match_soundcard(id, microphones, include_loopback)['id'])
+    return _Microphone(id=_match_soundcard(str(id), microphones, include_loopback)['id'])
```

Comment:
I think the case should not happen here, but later on in `_match_soundcard`. Unless I am misunderstanding something.

Reply:
You are correct. The crash happens on when it tries to pattern match the int to a str within the if statements.

This mitigation is actually completely pointless now. As with the "naive try return" it will return iff it's an int and it's indexed. Only strings or larger numbers will fail(#125 (comment)). I've removed it for now.

If this bug comes back in the future simply put "id = str(id)" at the start of _match_soundcard().